vulnerabilities/importers/apache_httpd.py

```diff
@@ -49,47 +54,81 @@ def updated_advisories(self):
         # skips processing the response further to avoid duplicate work
 
         if create_etag(data_src=self, url=self.url, etag_key="ETag"):
-            data = fetch_xml(self.url)
-            advisories = to_advisories(data)
+            links = fetch_links(self.url)
+            advisories = []
+            for link in links:
+                data = requests.get(link).json()
+                advisories.append(self.to_advisory(data))
             return self.batch_advisories(advisories)
 
         return []
 
-
-def to_advisories(data):
-    advisories = []
-    for issue in data:
-        resolved_packages = []
-        impacted_packages = []
-        for info in issue:
-            if info.tag == "cve":
-                cve = info.attrib["name"]
-
-            if info.tag == "title":
-                summary = info.text
-
-            if info.tag == "fixed":
-                resolved_packages.append(
-                    PackageURL(type="apache", name="httpd", version=info.attrib["version"])
-                )
-
-            if info.tag == "affects" or info.tag == "maybeaffects":
-                impacted_packages.append(
-                    PackageURL(type="apache", name="httpd", version=info.attrib["version"])
-                )
-
-        advisories.append(
-            Advisory(
-                vulnerability_id=cve,
-                summary=summary,
-                impacted_package_urls=impacted_packages,
-                resolved_package_urls=resolved_packages,
+    def to_advisory(self, data):
+        cve = data["CVE_data_meta"]["ID"]
+        descriptions = data.get("description", {}).get("description_data", [])
+        description = None
+        for desc in descriptions:
+            if desc.get("lang") == "eng":
+                description = desc.get("value")
+                break
+
+        impacts = data.get("impact", [])
+        impact = None
+        for imp in impacts:
+            value = imp.get("other")
+            if value is not None:
+                impact = value
+                break
+
+        if impact is not None:
+            severity = VulnerabilitySeverity(
+                system=scoring_systems["apache_httpd"],
+                value=impact,
+            )
+            reference = Reference(
+                reference_id=cve,
+                url=self.ref_url.format(cve),
+                severities=[severity],
+            )
+        else:
+            reference = Reference(
+                reference_id=cve,
+                url=self.ref_url.format(cve),
             )
```

Comment:
This big code block can be avoided by doing something like : 

```python
severities = []
if impact:
  severities.append( VulnerabilitySeverity(......))

reference = Reference(...... severities=severities)

```

Reply:
```python
severities = []
if impact is not None:
    severities.append(
        VulnerabilitySeverity(
            system=scoring_systems["apache_httpd"],
            value=impact,
        )
    )
reference = Reference(
    reference_id=cve,
    url=self.ref_url.format(cve),
    severities=severities,
)
```